### PR TITLE
autoconf-archive: get it working on Windows

### PR DESCRIPTION
**Pull request by madebr**

### CODE REVIEW

File: recipes/autoconf-archive/all/conanfile.py [hunk 1]
```diff
@@ -47,10 +47,13 @@ def package(self):
...
+                     os.path.join(self.package_folder, "res", "aclocal"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
 
     def package_info(self):
-        aclocal_path = os.path.join(self.package_folder, "bin", "share", "aclocal").replace("\\", "/")
-        self.output.info("Appending ACLOCAL_PATH environment var: {}".format(aclocal_path))
-        self.env_info.ACLOCAL_PATH.append(aclocal_path)
+        aclocal_path = tools.unix_path(os.path.join(self.package_folder, "res", "aclocal"))
+        self.output.info("Appending AUTOMAKE_CONAN_INCLUDES environment var: {}".format(aclocal_path))
+        self.env_info.AUTOMAKE_CONAN_INCLUDES.append(aclocal_path)
```
Comment (madebr) [inline recipes/autoconf-archive/all/conanfile.py:59]: Looks like we're escaping `AUTOMAKE_CONAN_INCLUDES` correctly, but not `ACLOCAL_PATH` on Windows.

Comment (madebr) [reply recipes/autoconf-archive/all/conanfile.py:59]: Ideally, `AUTOMAKE_CONAN_INCLUDES` is not necessary and we use `ACLOCAL_PATH` 100%.

Comment (sparik) [reply recipes/autoconf-archive/all/conanfile.py:59]: Can you clarify? I don't quite understand this. `ACLOCAL_PATH` is needed for the macros to be in the search path. What's the purpose of `AUTOMAKE_CONAN_INCLUDES`? Where can I read more about that?

Comment (madebr) [reply recipes/autoconf-archive/all/conanfile.py:59]: `AUTOMAKE_CONAN_INCLUDES` is added by a conan patch (before I knew the purpose of `ACLOCAL_PATH` myself.
Its functionality is the same as `ACLOCAL_PATH`, but better supported apparently.

The only place you can read about it, is in the patches subfolder of the automake recipe.

Comment (sparik) [reply recipes/autoconf-archive/all/conanfile.py:59]: understood, thanks

**Pull request merged**